lib/client.js

```diff
@@ -53,8 +54,12 @@ function Client(config = {}) {
   if (config.version) {
     this.version = config.version;
   }
-
-  this.connectionSettings = cloneDeep(config.connection || {});
+  // if the type of connection settings is a function, pass it to the next layer
+  if (typeof config.connection === 'function') {
```

Comment:
_.isFunction is preferable

Reply:
https://github.com/tgriesser/knex/pull/3403/files#diff-cbc1ae6887774cf34c8d4bc73dece12fR11